Don't propagate server.port

ericbottard opened the pull request:
Fixes spring-cloud/spring-cloud-deployer-cloudfoundry#46

markfisher commented on src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java:
```diff
@@ -15,10 +15,10 @@
 package org.springframework.cloud.deployer.spi.cloudfoundry;
 
-import static java.lang.Integer.parseInt;
+import static java.lang.Integer.*;
```
update formatting rules to avoid wildcard imports

markfisher commented on src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java:
```diff
@@ -100,58 +100,67 @@ public String deploy(AppDeploymentRequest request) {
+			// Remove server.port as CF assigns a port for us, and we don't want to override that
+			String port = properties.remove("server.port");
+			if (port != null) {
+				logger.info(String.format("Ignoring server.port=%s property for app %s, as Cloud Foundry will assign a dynamic port", name, port));
```
INFO might be fine, but should we consider WARN? cc @markpollack

markpollack replied:
I'd make it WARN since it is something one should ideally take action on to remove from the stream definition.

markpollack replied:
The log message isn't quite right, the individual app will get a dynamic port, but the route to the app - which his what is exposed to the public - will be over port 80, we should indicate that instead.  Agreed?

ericbottard replied:
Addressed review comments

markfisher commented:
lots of formatting changes make this diff hard to follow... I assume the only actual changes are within lines 103-108?

ericbottard commented:
> lots of formatting changes make this diff hard to follow... I assume the only actual changes are within lines 103-108?

yes.
Grrr... I just updated those (my formatting prefs) because Janne made me realize I lost them (on IDEA upgrade, I assume). I'm using the `spring-framework` scheme I always used. Maybe the code was formatted differently before.
Anyhow, I fixed the statics and force pushed

ericbottard commented:
Closing in favor of https://github.com/spring-cloud/spring-cloud-deployer-cloudfoundry/pull/78
